spec/inputs/integration/jms_spec.rb

```diff
@@ -164,21 +162,80 @@
       end
 
-      context 'when neither header nor property is skipped ' do
-        it 'should process properties, headers and the message' do
+      context 'when include_headers => false' do
+        let (:config) { super().merge('include_headers' => 'false') }
+
+        it 'should skip all headers' do
           send_message do |session|
             msg = session.message(message)
-            msg.set_string_property('this', 'this_prop')
-            msg.set_string_property('that', 'that_prop')
-            msg.set_string_property('the_other', 'the_other_prop')
+            msg.reply_to = session.create_destination(:topic_name => SecureRandom.hex(8))
+            msg.set_string_property('some', 'property')
             msg
           end
-          expect(queue.first.get('message')).to eql (message)
-          expect(queue.first.get('jms_timestamp')).to_not be_nil
-          expect(queue.first.get('jms_destination')).to_not be_nil
-          expect(queue.first.get('this')).to eq('this_prop')
-          expect(queue.first.get('that')).to eq('that_prop')
-          expect(queue.first.get('the_other')).to eq('the_other_prop')
+          event = queue.first.to_hash_with_metadata
+          expect( event.keys.find { |name| name.start_with?('jms_') } ).to be nil
+        end
+      end
+
+      context 'when include_header => false (deprecated)' do
+        let (:config) { super().merge('include_header' => 'false') }
+
+        it 'should skip all headers' do
+          send_message do |session|
+            msg = session.message(message)
+            msg.reply_to = session.create_destination(:topic_name => SecureRandom.hex(8))
+            msg.set_string_property('some', 'property')
+            msg
+          end
+          event = queue.first.to_hash_with_metadata
+          expect( event.keys.find { |name| name.start_with?('jms_') } ).to be nil
+        end
+      end
+
+      context 'when neither header nor property is skipped', :ecs_compatibility_support do
+        ecs_compatibility_matrix(:disabled, :v1, :v8) do |ecs_select|
+
+          let(:ecs_compatibility?) { ecs_select.active_mode != :disabled }
+
+          let (:config) { super().merge('ecs_compatibility' => ecs_select.active_mode) }
+
+          it 'should process properties, headers and the message' do
+            send_message do |session|
+              msg = session.message(message)
+              msg.set_string_property('this', 'this_prop')
+              msg.set_int_property('camelCase', 42)
+              msg.set_boolean_property('JMSFlag', true)
+              msg
+            end
+
+            event = queue.first
+
+            expect(event.get('message')).to eql(message)
+
+            # headers
+            if ecs_compatibility?
+              expect(event.include?('jms_timestamp')).to be false
+              expect(event.include?('jms_destination')).to be false
+              expect(event.get('[@metadata][input][jms][headers][jms_timestamp]')).to be_a Integer
```

Comment:
usability nitpick: do these need the `jms_` prefix when they are put into a JMS-specific namespace by default?

Reply:
they do not necessarily need the `jms_` prefix but than we need to document precisely what is supported and how header names change with `ecs_compatibility` ...

one of the names look weird (`jms_delivery_mode_sym`) due the decision to copy what the jruby-jms library decided rather than taking these from JMS, here's the list: https://github.com/reidmorrison/jruby-jms/blob/master/lib/jms/message.rb#L76-L85

do we want to re-invent these header names as part of the ECS effort?
the JMS standard uses the JMS prefix e.g. [`JMSMessageID` header field](https://docs.oracle.com/javaee/7/api/javax/jms/Message.html#getJMSMessageID--) the current `jms_message_id` name matches the convention, in theory there might be provider specific headers that could be extracted using some extra APIs and than having the `jms_` naming prefix is useful.

maybe just fixing the `jms_delivery_mode_sym` to be simply `jms_delivery_mode` while keeping the header value?

Reply:
I'm +1 to follow the standard, even if looks a little redundant with our namespacing.
I'm also +1 to droping the `_sym` suffix -- do we need to also stringify its value or anything? I assume that it is a `Symbol` from the name, but am not seeing any specs around it.

Reply:
okay, done. we'll be doing an explicit, very similar, mapping as we got from the JMS gem except:
- new `jms_delivery_mode` and mapping it to a name (just like `jms_delivery_mode_sym`)
- `jms_delivery_mode_sym` only kept in legacy mode
- headers such as `jms_correlation_id` only set when present (no longer set a `nil`)
- `jms_reply_to` fixed to do the `Destination#toString` instead of setting the JMS object 
  (has caused issues before - we also provide a work-around to skip the header in the docs)